### ISSUE TITLE
[Enhancement] Modify date component to contain calendar icon

### ISSUE THREAD

**Issue by NillerW** [kirbydesign/designsystem#3234]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement

The Kirby input-field of type date should support an icon.

This will enable the to-be-implemented functionality of selecting date by clicking on the icon.

## Describe the solution you'd like

The input-field should allow an icon inside the date input.

- [x] Icon should be placed to the right-most side, such that it aligns with the rest of form-field affix implementations.
- [x] Icon should by default be a calendar icon
- [ ] ~~It should be possible to change the default icon~~ Moved out of scope
- [x] Icon should be clickable to allow the browsers built-in date picker (#3235)
- [ ] ~~Must work with `inputmask` plugin~~
  - We suggest to remove the inputmask directive from the `type=date` input. There are quite a few bugs around inputmask and a more robust solution could be to remove the native calendar icon and overlay a kirby calendar icon

### Tasks (refinement 6/12 2023)

- [x] Decision: Utilize *affix* ~~vs. new, but similar solution~~
- [ ] Prevent use of *prefix* with `type="date"` - or handle it properly
- [ ] Position custom icon on top of browser's icon
  - [x] Handle padding: If `type="date"` then use only initial/default padding (do not add width of slotted element to padding)
  - [x] Use `pointer-events: none`
  - [x] Use **_default_** background color to cover browser's icon - other theme colors than default is out of scope
- [x] Handle interaction states (_hover_ and _focus_) for default theme color
- [ ] May need further refinement: Decide what to do with `inputmask`

## Are there any additional context?
This is to enable the creation of a date range component. 

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

________

(fix the date mask issue for dateInput - not allowing affix)

**Comment by jkaltoft** [comment]
# Just-before-Christmas status

Screenshots and screen recordings

- Firefox Nightly v122
- Firefox v120
- Google Chrome Canary v122
- Google Chrome v120
- Microsoft Edge Dev v121 (language set to danish on OS level)
- Safari iOS 17.0 iPhone 15 Pro
- Safari Technology Preview v17.4
- Safari v17.1.2

## Firefox Nightly v122

https://github.com/kirbydesign/designsystem/assets/150451/f4fed1dc-9b0f-4987-ad7e-814ddddbd8b3

![Firefox Nightly v122](https://github.com/kirbydesign/designsystem/assets/150451/bb46ee7e-98e6-4750-a983-93c7706b7253)

## Firefox v120

https://github.com/kirbydesign/designsystem/assets/150451/0a0593b1-290f-409f-a513-1e318eb18076

![Firefox v120 1](https://github.com/kirbydesign/designsystem/assets/150451/d2101a52-eca4-43ae-a425-0026cad08621)

![Firefox v120 2](https://github.com/kirbydesign/designsystem/assets/150451/4ace92f4-dc10-45f9-ae32-f21f880ac4d5)

![Firefox v120](https://github.com/kirbydesign/designsystem/assets/150451/b2288c22-2a58-441f-988c-b06d042c0e69)

## Google Chrome Canary v122

https://github.com/kirbydesign/designsystem/assets/150451/146b1f20-7385-43da-a9ec-b74041240291

![Google Chrome Canary v122](https://github.com/kirbydesign/designsystem/assets/150451/d0017432-0226-4e00-9e1a-724599e2c650)

## Google Chrome v120

https://github.com/kirbydesign/designsystem/assets/150451/315fa526-5179-482d-92f2-ab1c920e6ec2

![Google Chrome v120](https://github.com/kirbydesign/designsystem/assets/150451/ab5bff90-40e3-4d94-b702-fc3a46e21b5f)

## Microsoft Edge Dev v121

Language set to danish on OS level

https://github.com/kirbydesign/designsystem/assets/150451/95af7205-c328-45a1-9f02-d10d717b142b

![Microsoft Edge Dev v121](https://github.com/kirbydesign/designsystem/assets/150451/973a4750-3fdb-454d-90df-35299b37cf10)

## Safari iOS 17.0 iPhone 15 Pro (Simulator)

https://github.com/kirbydesign/designsystem/assets/150451/45234fb8-4fff-4189-b9a0-714e25e490a9

<table>
    <tr>
        <td>

![iOS 17 0 iPhone 15 Pro 1](https://github.com/kirbydesign/designsystem/assets/150451/0348fd71-10a6-4a4a-8373-fb9d9cf2fe78)

</td>
        <td>

![iOS 17 0 iPhone 15 Pro 2](https://github.com/kirbydesign/designsystem/assets/150451/dcf98d41-976d-4b58-ab1d-841c41de1acd)

</td>
        <td>

![iOS 17 0 iPhone 15 Pro 3](https://github.com/kirbydesign/designsystem/assets/150451/29067f13-9edf-473e-be90-803829160a05)

</td>
    </tr>
</table>

## Safari Technology Preview v17.4

https://github.com/kirbydesign/designsystem/assets/150451/28dd1047-e41c-48a5-8c7b-36a6bea640d9

![Safari Technology Preview v17 4](https://github.com/kirbydesign/designsystem/assets/150451/f56c9ea2-358c-4c0c-baa8-71ff9d810066)

## Safari v17.1.2

https://github.com/kirbydesign/designsystem/assets/150451/21b8ee65-6355-4d8d-afd1-3e185e6ef3bb

![Safari v17 1 2](https://github.com/kirbydesign/designsystem/assets/150451/a0e41539-dc3f-420b-adfa-e949bee41d2c)

**Comment by jkaltoft** [comment]
@jakobe @NillerW 

Regarding the requirement "It should be possible to change the default icon".

What is the use case for changing the icon for the date picker to something other than the calendar icon? I talked to @RasmusKjeldgaard and @henrikvoetmand about this and we couldn't figure out why this should be possible.

I may be missing some context. Can you elaborate on this?

**Comment by NillerW** [comment]
I don't recall if there was a specific use-case - I believe there wasn't and currently isn't. 

It was more to have the freedom to change it, should the need arise. I am all for, "then let's use the same icon everywhere", but that's hardly ever the case. 

I could see scenarios where a calendar icon could be filled with another icon, like birth, death, payday etc. simply for user experience. 

It's up to @henrikvoetmand 

Obviously it could be added later, but might be way easier and cheaper to just do it now while you're at it.
